data/binding/preference.go

```diff
@@ -22,6 +22,21 @@ func BindPreferenceBool(key string, p fyne.Preferences) Bool {
 		})
 }
 
+// BindPreferenceBoolList returns a bound list of bool values that is managed by the application preferences.
+// Changes to this value will be saved to application storage and when the app starts the previous values will be read.
+//
+// Since: 2.6
+func BindPreferenceBoolList(key string, p fyne.Preferences) BoolList {
```

Comment:
Should we really fill the API with new functions for each type? We could do a single `BindPreferenceList[T bool | float64 | int | string]()` and potentially expand that typeset in the future.

Reply:
See note in slack, we might even be able to just do `[T comparable]` and match all comparable types.

Reply:
I would rather complete the API as it exists already (as it was basically missing). We could have a generic version but as I doubt other items will be added in the future I don't think it would be pertinent to add a new pattern for this PR.

Reply:
Yeah, I kind of agree. In general, I think we should focus on using generics now in these cases where it makes a lot of sense over flooding the API surface with a bunch of new functions but like you say, it should have been there from the start. Will approve 👍